### PR TITLE
Document the general Gamify-IT architecture and fix two ReadTheDocs warnings

### PR DESCRIPTION
**Pull request by delvh**
The additional component diagram might be less formally correct, but I think it is far more understandable.

Part of Gamify-IT/issues#333

### CODE REVIEW

**Review by robojones — APPROVED** [review]

File: dev-manuals/architecture/general-architure.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+In general, _Gamify-IT_ can be split up into three main components: the _Overworld_, the _Lecturer Interface_, and the _Minigames_.  
+Purpose:
+
+- minigame: A small game to learn something by playing
+- overworld: consists of areas you can walk around in, interact with your environment or start minigames
+- lecturer interface: The allmighty configurator to configure what happens where in the overworld, or to set what is displayed in the minigames.
+
+Every of these services has its own backend, except the lecturer interface whose backend is the overworld backend as well.
+
```
Comment (robojones) [inline dev-manuals/architecture/general-architure.md:15]: Please add a link to the service's developer manual here (e.g. <https://gamifyit-docs.readthedocs.io/en/latest/dev-manuals/index.html>).

File: dev-manuals/architecture/general-architure.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
+# General architecture
+
```
Comment (robojones) [inline dev-manuals/architecture/general-architure.md:2]: This page could also use a Table of Contents (can be manually created or generated with InteliJ)

**Review by robojones — APPROVED** [review]

**Comment by robojones** [comment]
I just noticed that the filename `dev-manuals/architecture/general-architure.md` is misspelled

**Comment by delvh** [comment]
Me as well

**Review by Gr33ndev — APPROVED** [review]

**Pull request merged**